python/GafferSceneUI/LightEditor.py

```diff
@@ -408,27 +389,35 @@ def __toggleBoolean( self, inspections ) :
 		) :
 			return False
 
-		soleValue = sole( values )
+		valueData = [ i.value() for i in inspections ]
+		soleValue = sole( v.value if v is not None else False for v in valueData )
```

Comment:
I haven't followed the logic through to see why, but this is still not matching the spreadsheet behaviour :

1. Select two cells, where one is muted and one is not.
2. Hit <kbd>Enter</kbd> to toggle. They each flip their current value, whereas the Spreadsheet flips them all to the same value.

Reply:
I added some tests for this case in https://github.com/GafferHQ/gaffer/pull/4948/commits/a502068ef81b71bf1572e9355efa9181f721de8f and fixed it in https://github.com/GafferHQ/gaffer/pull/4948/commits/0776038e2bc8e94d18050b0b588105fa264b5c9a

There are a couple other tests / fixes for another case that I came across today. Prior to this commit, if you toggled a light and its parent at the same time, the enabled state would not be correct. I was getting the parent value from pre-toggle state, which means it would be opposite of the post-toggle state if the parent and child are both toggled.